protostar/utils/data_transformer_test.py

```diff
@@ -0,0 +1,229 @@
+from typing_extensions import Literal
+
+import pytest
+
+from .data_transformer import (
+    DataTransformerException,
+    PythonData,
+    CairoData,
+    from_python_transformer,
+    from_python_events_transformer,
+    to_python_transformer,
+    to_python_events_transformer,
+)
+
+TEST_ABI = [
+    {
+        "members": [
+            {"name": "felt_field", "offset": 0, "type": "felt"},
+            {"name": "inner_field", "offset": 1, "type": "InnerStruct"},
+        ],
+        "name": "Test",
+        "size": 4,
+        "type": "struct",
+    },
+    {
+        "members": [
+            {"name": "a", "offset": 0, "type": "felt"},
+            {"name": "b", "offset": 1, "type": "felt"},
+            {"name": "c", "offset": 2, "type": "felt"},
+        ],
+        "name": "InnerStruct",
+        "size": 3,
+        "type": "struct",
+    },
+    {
+        "members": [
+            {"name": "low", "offset": 0, "type": "felt"},
+            {"name": "high", "offset": 1, "type": "felt"},
+        ],
+        "name": "Uint256",
+        "size": 2,
+        "type": "struct",
+    },
+    {
+        "inputs": [{"name": "amount", "type": "felt"}],
+        "name": "increase_balance",
+        "outputs": [],
+        "type": "function",
+    },
+    {
+        "inputs": [],
+        "name": "get_balance",
+        "outputs": [{"name": "res", "type": "felt"}],
+        "stateMutability": "view",
+        "type": "function",
+    },
+    {
+        "inputs": [
+            {"name": "arg", "type": "Test"},
+            {"name": "felt_arg", "type": "felt"},
+            {"name": "uint256_arg", "type": "Uint256"},
+            {"name": "felt_list_arg_len", "type": "felt"},
+            {"name": "felt_list_arg", "type": "felt*"},
+            {"name": "inst_list_arg_len", "type": "felt"},
+            {"name": "inst_list_arg", "type": "InnerStruct*"},
+        ],
+        "name": "constructor",
+        "outputs": [],
+        "type": "constructor",
+    },
+    {
+        "data": [
+            {"name": "arg", "type": "Test"},
+            {"name": "felt_arg", "type": "felt"},
+            {"name": "uint256_arg", "type": "Uint256"},
+            {"name": "felt_list_arg_len", "type": "felt"},
+            {"name": "felt_list_arg", "type": "felt*"},
+            {"name": "inst_list_arg_len", "type": "felt"},
+            {"name": "inst_list_arg", "type": "InnerStruct*"},
+        ],
+        "name": "constructor_event",
+        "type": "event",
+        "from_address": "0",
+    },
+    {
+        "data": [{"name": "input", "type": "(felt, felt)"}],
+        "name": "tuple_event",
+        "type": "event",
+        "from_address": "0",
+    },
+]
```

Comment:
Can you create a test utility function that generates the ABI? It would make the test suite more readable and maintainable.

 e.g.
```python
abi = create_abi(
"""\
@constructor
func 

# ...
"""
)
```

Also, you split ABI/PYTHON_INPUT/CAIRO_INPUT into triplets grouped by the input type.
arg, felt_arg, uint256, etc.

Reply:
done